core/src/main/java/com/google/cloud/sql/nativeimage/CloudSqlFeature.java

```diff
@@ -0,0 +1,127 @@
+          access.findClassByName("[Lcom.zaxxer.hikari.util.ConcurrentBag$IConcurrentBagEntry;"));
+
+      RuntimeReflection.register(access.findClassByName("[Ljava.sql.Statement;"));
+    }
+
+    // Register PostgreSQL driver config.
+    if (access.findClassByName(POSTGRES_SOCKET_CLASS) != null) {
+      NativeImageUtils.registerClassForReflection(
+          access, "com.google.cloud.sql.postgres.SocketFactory");
+      NativeImageUtils.registerClassForReflection(access, "org.postgresql.PGProperty");
+    }
+
+    // Register MySQL driver config.
+    if (access.findClassByName(MYSQL_SOCKET_CLASS) != null) {
+      NativeImageUtils.registerClassForReflection(access, MYSQL_SOCKET_CLASS);
+
+      NativeImageUtils.registerClassForReflection(access, "com.mysql.jdbc.StandardSocketFactory");
+
+      NativeImageUtils.registerConstructorsForReflection(
+          access, "com.mysql.cj.conf.url.SingleConnectionUrl");
+
+      // for mysql-j-5
+      NativeImageUtils.registerConstructorsForReflection(
+          access, "com.mysql.jdbc.log.StandardLogger");
+      // for mysql-j-8
+      NativeImageUtils.registerConstructorsForReflection(access, "com.mysql.cj.log.StandardLogger");
+
+      Class<?> cjExceptionClass = access.findClassByName("com.mysql.cj.exceptions.CJException");
+      if (cjExceptionClass != null) {
+        // The CJException exists only jdbc/mysql-j-8 module's dependency
+        access.registerSubtypeReachabilityHandler(
+            (duringAccess, exceptionClass) ->
+                NativeImageUtils.registerClassForReflection(duringAccess, exceptionClass.getName()),
+            cjExceptionClass);
+      }
+
+      Class<?> mySqlNonTransientConnectionException =
+          access.findClassByName(
+              "com.mysql.jdbc.exceptions.jdbc4.MySQLNonTransientConnectionException");
+      if (mySqlNonTransientConnectionException != null) {
+        NativeImageUtils.registerConstructorsForReflection(
+            access, mySqlNonTransientConnectionException.getName());
+      }
+
+      Class<?> mySqlNonTransientException =
+          access.findClassByName("com.mysql.jdbc.exceptions.MySQLNonTransientException");
+      if (mySqlNonTransientException != null) {
+        NativeImageUtils.registerConstructorsForReflection(
+            access, mySqlNonTransientException.getName());
+      }
+
+      // JDBC classes create socket connections which must be initialized at run time.
+      RuntimeClassInitialization.initializeAtRunTime("com.mysql.cj.jdbc");
+
+      // Additional MySQL resources.
+      resourcesRegistry.addResourceBundles(
+          ConfigurationCondition.alwaysTrue(), "com.mysql.cj.LocalizedErrorMessages");
+      resourcesRegistry.addResourceBundles(
+          ConfigurationCondition.alwaysTrue(), "com.mysql.jdbc.LocalizedErrorMessages");
+    }
+
+    // https://github.com/netty/netty/issues/11638
+    Class<?> bcsslEngine = access.findClassByName("org.bouncycastle.jsse.BCSSLEngine");
```

Comment:
I think this should actually be `io.netty.handler.ssl.BouncyCastleAlpnSslUtils`, if we're following the example in the linked issue

Reply:
Thanks. Trying that.